### PR TITLE
Add NTP server capability to DS3231 driver

### PR DESCRIPTION
**Pull request by pkkrusty**
## Description:

**Related issue (if applicable):** fixes #14293

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with Tasmota core ESP32 V.2.0.2
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/support_features.ino [hunk 1]
```diff
@@ -787,8 +787,9 @@ void ResponseAppendFeatures(void)
 #ifdef USE_SHIFT595
     feature8 |= 0x00080000;  // xdrv_60_shift595.ino
 #endif
-
-//    feature8 |= 0x00100000;
+#if defined(USE_I2C) && defined(USE_DS3231)
```
Comment (arendst) [inline tasmota/support_features.ino:790]: Duplicate. Is already in this file. Pls remove.

Next time use a global search within VSC to track any name.

Comment (pkkrusty) [reply tasmota/support_features.ino]: Done.

Comment (pkkrusty) [reply tasmota/support_features.ino]: If this will get approved, I'll start working on documentation.

**Comment by arendst** [comment]
It's a nice exercise in copy and paste.

There are some changes possible to cut down on code size.

To continue the exercise perhaps you can come up with a smaller footprint although I have to agree the extra 700 bytes used now can possibly only be decreased to 600 bytes.

In the "old" days of Tasmota that was a big issue. For me it's still but others disagree.

**Comment by pkkrusty** [comment]
Yeah, it's entirely copy/paste, but I don't know enough to consolidate the code. I could make variable names shorter...

**Comment by arendst** [comment]
No that makes no difference.

I understand your situation. I'll merge this while you try to fiddle around with VSC to make the code footprint smaller whenever you're up to it. I have to admit the source wasn't written in a small code footprint either.

**Pull request merged**